### PR TITLE
Rewrite the "word by word" section

### PR DESCRIPTION
**Pull request by pavpanchekha**
Several students noted that this section was hard to follow, both because it was unclear where the code is supposed to go and also because there was a lot of text describing a lot of code and it was hard to match one with the other.

This PR breaks the section up into pieces, fixing #1126. The before/after is below: the section got longer, but I think that's fine because it was a short section to begin with:

Screenshot 2024-03-08 at 2 13 22 PM
Screenshot 2024-03-08 at 2 13 28 PM

### CODE REVIEW

File: book/text.md [hunk 1]
```diff
@@ -298,35 +298,47 @@ a time:[^whitespace]
     including for identifying word boundaries.
 
 ``` {.python expected=False}
```
Comment (chrishtr) [inline book/text.md:300]: Why expected=false?

File: book/text.md [hunk 2]
```diff
@@ -298,35 +298,47 @@ a time:[^whitespace]
...
+    # ...
+    for word in text.split():
+        # ...
+    return display_list
+```
+
+Unlike Chinese characters, words are different sizes, so we need to
+measure the width of each word:
+
+``` {.python expected=False }
```
Comment (chrishtr) [inline book/text.md:311]: Why expected=false?

File: book/text.md [hunk 3]
```diff
@@ -298,35 +298,47 @@ a time:[^whitespace]
...
+        w = font.measure(word)
+    # ...
+```
+
+Here I've chosen to use Tk's default font. Now, if we draw the text at
+`cursor_x`, its right end would be at `cursor_x + w`. That might be
+past the right edge of the page, and in this case we need to make
+space by wrapping to the next line:
+
+``` {.python expected=False}
```
Comment (chrishtr) [inline book/text.md:325]: Why expected=false?

File: book/text.md (hunk 2 above)
Comment (pavpanchekha) [reply book/text.md:311]: Same as above; the `for` loop moves elsewhere and the `font` line changes when we add styling so there's not much left to match.

File: book/text.md (hunk 3 above)
Comment (pavpanchekha) [reply book/text.md:325]: Similar to above, though at least in principle we could make this match _something_ by dropping the cursor_x/y lines (which move into `flush`) and at least matching the if statement.
